File: progressBar/java/ProgressBar/src/bar/ProgressBar.java
package bar;

/**
 * a simple commandline progress bar
 * @author philip
 *
 */
public class ProgressBar {

	private int capacity; // capacity of the bar, i.e., what value constitutes 100%
	private int size; // length of the bar, i.e., how many characters it is comprised of
	
	public ProgressBar (int capacity, int size) {
		this.capacity = capacity;
		this.size = size;
		this.update(0);
	}
	
	/**
	 * update the progress bar with a new value
	 * @param progress new value for the progress
	 */
	public void update(double progress) {
		// calculate the necessary number of "#" to add to the bar
		double marks = (progress/this.capacity) * this.size;
		
		// build the bar
		StringBuffer buffer = new StringBuffer();
		buffer.append("[");

		for(int i = 1; i <= marks; i++) {
			buffer.append("#");
		}
		for(int i = this.size; i > marks; i--) {
			buffer.append(" ");
		}
		buffer.append("] " + (int)progress + "/" + this.capacity + "\r");
		System.out.print(buffer.toString());
	}
	
	/**
	 * when done, call this to move the cursor to a new line
	 */
	public void finish() {
		System.out.println();
	}
}

File: progressBar/python/ProgressBar.py
# coding: utf-8

import sys
import math

class ProgressBar(object):
	def __init__ (self, capacity, size):
		"""
		Initialize the progress bar.
		capacity: maximum capacity of bar, e.g., what is 100%
		size: length of the bar, e.g., how many characters it's made up of
		"""
		self._capacity = capacity
		self._size = size
		self.update(0)

	def update (self, value):
		"""
		update the progress bar with a new value.
		value: updated progress
		"""
		marks = math.floor((value/self._capacity) * self._size)
		line = "["
		line += "#" * marks
		line += " " * (self._size - marks)
		line += "] " + str(value) + "/" + str(self._capacity) + "\r"
		sys.stdout.write(line)

	def finish (self):
		"""
		puts the cursor on a new line; call when done with updating the bar
		"""
		print()
